File: public/docs/_examples/intro/dart/lib/hero_list_component.dart
library developer_guide_intro.hero_list_component;

import 'package:angular2/angular2.dart';
import 'package:developer_guide_intro/hero.dart';
import 'package:developer_guide_intro/hero_detail_component.dart';
import 'package:developer_guide_intro/hero_service.dart';

@Component(
    selector: 'hero-list',
    templateUrl: 'hero_list_component.html',
    directives: const [CORE_DIRECTIVES,HeroDetailComponent],
    providers: const [HeroService])
class HeroListComponent {
  List<Hero> heroes;
  Hero selectedHero;

  HeroListComponent(HeroService heroService){
    heroes = heroService.getHeroes();
  }

  selectHero(Hero hero){
    selectedHero = hero;
  }

}

File: public/docs/_examples/intro/ts/src/app/hero-list.component.ts
// #docplaster

import {Component} from 'angular2/core';
import {Hero} from './hero';
import {HeroDetailComponent} from './hero-detail.component';
import {HeroService} from './hero.service'

// #docregion metadata
// #docregion providers
@Component({
// #enddocregion providers
  selector:    'hero-list',
  templateUrl: 'app/hero-list.component.html',
  directives:  [HeroDetailComponent],
// #docregion providers
  providers:   [HeroService]
})
// #enddocregion providers
// #enddocregion metadata
/*
// #docregion metadata, providers
export class HeroesComponent { ... }
// #enddocregion metadata, providers
*/
// #docregion class
export class HeroListComponent {
// #docregion ctor
  constructor(service: HeroService) {
    this.heroes = service.getHeroes();
  }
// #enddocregion ctor

  heroes:Hero[];
  selectedHero:Hero;
  selectHero(hero: Hero) { this.selectedHero = hero; }
}
// #enddocregion class
